components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -6104,11 +6104,14 @@ public API getLightweightAPIByUUID(String uuid, String requestedTenantDomain) th
      * @throws APIManagementException
      */
  
-    public API getLightweightAPI(APIIdentifier identifier) throws APIManagementException {
+    public API getLightweightAPI(APIIdentifier identifier, String orgId) throws APIManagementException {
 
         String uuid = null;
         try {
-            Organization org = new Organization(CarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+            if (orgId == null) {
+                orgId = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            }
```

Comment:
Lets try to avoid this and pass the tenant domain properly to the method from all the usages?

Reply:
https://github.com/wso2/carbon-apimgt/pull/9658/commits/142cd9cad67662fc64ac6e886127be649322fd43